src/ArborX_LinearBVH.hpp

```diff
@@ -45,6 +45,8 @@ struct PerThread
 namespace Details
 {
 struct HappyTreeFriends;
+template <typename Tree>
+struct CallbackWithDistance;
```

Comment:
This is absurd.  Just add an accessor to indexable getter to BVH and BF.
Or add to HappyTreeFriend if you have a good reason to think we should not do so.
FWIW `boost::geometry::index::rtree` has a `indexable_get()` member function.

Reply:
I introduced the accessor function. Named it `indexable_get()` following Boost, but not really a fan of it. Not sure what is a better name. `indexable_getter()` or `indexableGetter()`?